### PR TITLE
Sink `legacy.apt|yum.k8s.io` logs to bq for analysis

### PR DESCRIPTION
**Pull request by upodroid**

### CODE REVIEW

**Comment by upodroid** [comment]
/cc @xmudrii 

We haven't configured an google group that can access the bq datasets with PII info. We'll do that in a separate PR

File: infra/gcp/terraform/kubernetes-public/10-cluster-configuration.tf [hunk 1]
```diff
@@ -128,10 +128,6 @@ resource "google_container_cluster" "cluster" {
     }
   }
 
-  // Restrict master to Google IP space; use Cloud Shell to access
```
Comment (upodroid) [inline infra/gcp/terraform/kubernetes-public/10-cluster-configuration.tf:131]: Removed this feature as it wasn't helpful

File: infra/gcp/terraform/kubernetes-public/k8s-infra-prow.tf [hunk 2]
```diff
@@ -38,12 +38,6 @@ resource "google_service_account" "k8s_infra_prow" {
   display_name = local.prow_service_account
 }
 
-// Create a key for GCP Service Account k8s-infra-prow
```
Comment (upodroid) [inline infra/gcp/terraform/kubernetes-public/k8s-infra-prow.tf:41]: This is the key that was revoked by @ameukam recently as discussed privately. Cleanup the key and references to the secrets.

**Comment by upodroid** [comment]
This change has been applied

**Comment by upodroid** [comment]
```
 mahamed  MAHAALI-M-2PY9  ~  Desktop  Git  kubernetes   master  483⬆  1⚑  $  bq query --format=prettyjson --use_legacy_sql=false 'SELECT httpRequest,timestamp FROM `k8s-infra-public-pii.legacy_pkgs_k8s_io_logs.stdout_20240103` LIMIT 2' | jq
/Users/mahamed/google-cloud-sdk/platform/bq/bq.py:17: DeprecationWarning: 'pipes' is deprecated and slated for removal in Python 3.13
  import pipes
[
  {
    "httpRequest": {
      "cacheFillBytes": null,
      "cacheHit": null,
      "cacheLookup": null,
      "cacheValidatedWithOriginServer": null,
      "protocol": "HTTP/1.1",
      "referer": null,
      "remoteIp": "X.X.141.127",
      "requestMethod": "GET",
      "requestSize": "430",
      "requestUrl": "http://legacy.apt.k8s.io/",
      "responseSize": "344",
      "serverIp": "10.40.1.21",
      "status": "302",
      "userAgent": "Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2224.3 Safari/537.36"
    },
    "timestamp": "2024-01-03 17:52:45"
  },
  {
    "httpRequest": {
      "cacheFillBytes": null,
      "cacheHit": null,
      "cacheLookup": null,
      "cacheValidatedWithOriginServer": null,
      "protocol": "HTTP/1.1",
      "referer": null,
      "remoteIp": "X.X.141.127",
      "requestMethod": "GET",
      "requestSize": "455",
      "requestUrl": "http://legacy.apt.canary.k8s.io/",
      "responseSize": "696",
      "serverIp": "10.40.1.21",
      "status": "404",
      "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/36.0.1944.0 Safari/537.36"
    },
    "timestamp": "2024-01-03 17:52:45"
  }
]
```

**Review by xmudrii — APPROVED** [review]
Thank you for taking care of this!
/lgtm
/approve

**Pull request merged**